Integration/FourleafIntegration.php

```diff
@@ -104,7 +104,7 @@ public function doEnhancement(Lead $lead)
                 return false;
             }
 
-            $keys = $this->getDecryptedApiKeys();
+            $keys = $this->getKeys();
```

Comment:
What's the difference between getDecryptedApiKeys and getKeys, and what are the implications?

Why do fourleaf and xverify work today if this needs changing?

Reply:
getKeys is the prefered method. I was also trying to keep the code consistent. I believe everywhere else in this plugin getKeys is used.

Reply:
`    /**
     * Returns already decrypted keys.
     *
     * @return mixed
     */
    public function getKeys()
    {
        return $this->keys;
    }
`

Reply:
`    /**
     * Returns decrypted API keys.
     *
     * @param bool $entity
     *
     * @return array
     */
    public function getDecryptedApiKeys($entity = false)
    {
        static $decryptedKeys = [];

        if (!$entity) {
            $entity = $this->settings;
        }

        $keys = $entity->getApiKeys();

        $serialized = serialize($keys);
        if (empty($decryptedKeys[$serialized])) {
            $decrypted = $this->decryptApiKeys($keys, true);
            if (count($keys) !== 0 && count($decrypted) === 0) {
                $decrypted = $this->decryptApiKeys($keys);
                $this->encryptAndSetApiKeys($decrypted, $entity);
                $this->em->flush($entity);
            }
            $decryptedKeys[$serialized] = $this->dispatchIntegrationKeyEvent(
                PluginEvents::PLUGIN_ON_INTEGRATION_KEYS_DECRYPT,
                $decrypted
            );
        }

        return $decryptedKeys[$serialized];
    }
`